pkg/controller/secret/secretsync.go

```diff
@@ -182,6 +199,27 @@ func (c *SecretSyncController) translateSecret(cloudSecret *v1.Secret, cloudConf
 	return &secret, nil
 }
 
+// fetchEndpoint fetches the url provided against endpointkey
+func fetchEndpoint(endpointkey, conf, defaultEndpointValue string) (string, error) {
+	var re *regexp.Regexp
+	var match []string
+
+	// Fetch url provided against endpointkey
+	re = regexp.MustCompile(fmt.Sprintf("%s = (.*?)\n", endpointkey))
+	match = re.FindStringSubmatch(conf)
+	// If the url doesn't exist return default value
+	if len(match) <= 1 {
+		return defaultEndpointValue, nil
+	}
+
+	// If the url exists, validate it
+	if _, err := url.ParseRequestURI(match[1]); err != nil {
```

Comment:
In case of any error, can we log error and return default value? 
and lets not return any error from this method

Reply:
Done